src/components/dialog/Dialog.jsx

```diff
@@ -138,7 +156,9 @@ function BaseDialog({
   );
 
   const overlayClass = cx('js-dialog', 'sg-dialog__overlay', {
-    'sg-dialog__overlay--scroll': scroll === 'outside',
+    'sg-dialog__overlay--scroll':
+      (showScrollbarsInitially || hasFinishedTransition) &&
```

Comment:
@domidomi can we rename `showScrollbarsInitially` to something more meaningful? We can name it by reason or by effect but the next. condition is by reason and this one by effect. We shouldn't mix those 2 appraoches.